perimetre_asa/asaperimetre.py

```diff
@@ -29,16 +29,18 @@
 from qgis.core import QgsApplication
 from qgis.utils import iface
 from qgis.PyQt.QtCore import Qt
+from qgis.PyQt.QtWidgets import QMenu
 
 from .dock import AsaPerimetreDock
 from .processing.provider import AsaPerimetreProvider
 
 
 class AsaPerimetre:
 
-    def __init__(self):
+    def __init__(self, iface):
+        self.iface = iface
         self.provider = None
-        self.dock = None
+        self.dock = AsaPerimetreDock()
```

Comment:
Why creating a second dock here ?
There is already creation of the dock L53.

All GUI creation should stay in `initGui`, QGIS is not always launching the creation of the GUI, in case of `qgis_process` for instance

Reply:
It's ok fr this changes ? i can resolve ?